core/services/relay/evm/evm.go

```diff
@@ -75,18 +76,39 @@ func (r *Relayer) NewConfigProvider(args relaytypes.RelayArgs) (relaytypes.Confi
 
 type configWatcher struct {
 	utils.StartStopOnce
+	lggr             logger.Logger
 	contractAddress  common.Address
 	contractABI      abi.ABI
 	offchainDigester types.OffchainConfigDigester
 	configPoller     *ConfigPoller
 	chain            evm.Chain
+	new              bool
+	fromBlock        uint64
+	replayCtx        context.Context
+	replayCancel     context.CancelFunc
+	wg               sync.WaitGroup
 }
 
 func (c *configWatcher) Start(ctx context.Context) error {
+	if c.new {
+		// Only replay if its a brand new job.
+		c.wg.Add(1)
+		go func() {
+			defer c.wg.Done()
+			c.lggr.Infow("starting replay for config", "fromBlock", c.fromBlock)
+			if err := c.configPoller.destChainLogPoller.Replay(c.replayCtx, int64(c.fromBlock)); err != nil {
+				c.lggr.Errorf("error replaying for config", "err", err)
+			} else {
+				c.lggr.Infow("completed replaying for config", "fromBlock", c.fromBlock)
+			}
+		}()
+	}
 	return nil
 }
 
 func (c *configWatcher) Close() error {
+	c.replayCancel()
+	c.wg.Wait()
```

Comment:
we'll need the same logic to wrap these two lines here also? (if c.new && c.fromBlock!=0)

Reply:
no wait just returns immediately in that case